frontend/src/components/user/LessonLearnedDialog.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import Link from 'next/link'
+import { Dialog } from '@headlessui/react'
+import { MdWarningAmber } from 'react-icons/md'
+import { AiFillCloseCircle } from 'react-icons/ai'
+import { BsFillCheckCircleFill } from 'react-icons/bs'
+
+import DialogBox from './../DialogBox'
+import { classNames } from '~/utils/classNames'
+
+type Props = {
+  isOpen: boolean
+  closeModal: () => void
+  lessons: any[]
+}
+
+const LessonLearnedDialog: React.FC<Props> = (props) => {
+  const { isOpen, closeModal, lessons } = props
+  return (
+    <DialogBox isOpen={isOpen} closeModal={closeModal}>
+      <Dialog.Panel className={classNames(
+        'w-full max-w-2xl transform overflow-hidden rounded-2xl bg-white p-6',
+        'text-left align-middle shadow-xl transition-all'
+      )}>
+        <Dialog.Title
+          as="h3"
+          className="flex items-center space-x-3 text-xl font-bold leading-6 text-gray-800"
+        >
+          <span>Topics learned</span>
+        </Dialog.Title>
+        <div className="mt-4 divide-y">
+          <>
+            {lessons?.map((lesson, lIdx) => (
+              <div key={lIdx}>
+                {lesson?.answers.map((answer: any, index: number) => {
+                  const getCatTitle = answer?.question?.category?.title
+                  const is_correct = answer?.question?.choice_id == answer?.choice_id
```

Comment:
are `answers`, `questions`, and `category` nullable values? If not, then its okay to remove `?`. I think only `lessons` are nullable based on your Interface

```suggestion
                {lesson.answers.map((answer: any, index: number) => {
                  const getCatTitle = answer.question.category.title
                  const is_correct = answer.question.choice_id == answer.choice_id
```